Tests/FunctionalTests/NetworkResponseHandlerFunctionalTests.swift

```diff
@@ -318,6 +319,71 @@ class NetworkResponseHandlerFunctionalTests: FunctionalTestBase {
         XCTAssertTrue(storeResponsePayloadManager.getActivePayloadList().isEmpty)
     }
 
+    /// Tests that when an event is processed after a persisted reset event that the store payloads are saved
+    func testProcessResponseOnSuccess_afterPersistedResetEvent_savesStorePayloads() {
+        dataStore.set(key: EdgeConstants.DataStoreKeys.RESET_IDENTITIES_DATE, value: Date().timeIntervalSince1970) // date is before `event.timestamp`
+        networkResponseHandler = NetworkResponseHandler() // loads reset time on init
+
+        let event = Event(name: "test", type: "test-type", source: "test-source", data: nil)
+
+        networkResponseHandler.addWaitingEvents(requestId: "d81c93e5-7558-4996-a93c-489d550748b8",
+                                                batchedEvents: [event])
+
+        setExpectationEvent(type: FunctionalTestConst.EventType.EDGE, source: FunctionalTestConst.EventSource.RESPONSE_CONTENT, expectedCount: 1)
+        let jsonResponse = "{\n" +
+            "      \"requestId\": \"d81c93e5-7558-4996-a93c-489d550748b8\",\n" +
+            "      \"handle\": [" +
+            "           {\n" +
+            "            \"type\": \"state:store\",\n" +
+            "            \"payload\": [\n" +
+            "                {\n" +
+            "                    \"key\": \"s_ecid\",\n" +
+            "                    \"value\": \"MCMID|29068398647607325310376254630528178721\",\n" +
+            "                    \"maxAge\": 15552000\n" +
+            "                }\n" +
+            "            ]\n" +
+            "        }],\n" +
+            "      \"errors\": []\n" +
+            "    }"
+        networkResponseHandler.processResponseOnSuccess(jsonResponse: jsonResponse, requestId: "d81c93e5-7558-4996-a93c-489d550748b8")
+
+        // verify saved to store manager
+        let storeResponsePayloadManager = StoreResponsePayloadManager(EdgeConstants.DataStoreKeys.STORE_NAME)
+        XCTAssertFalse(storeResponsePayloadManager.getActivePayloadList().isEmpty)
+    }
+
+    /// Tests that when an event is processed before a persisted reset event that the store payloads are not saved
+    func testProcessResponseOnSuccess_beforePersistedResetEvent_doesNotSaveStorePayloads() {
+        let event = Event(name: "test", type: "test-type", source: "test-source", data: nil)
+
+        dataStore.set(key: EdgeConstants.DataStoreKeys.RESET_IDENTITIES_DATE, value: Date().timeIntervalSince1970) // date is after `event.timestamp`
```

Comment:
Same comment here about the dates being the same.  Could you set the RESET_IDENTITIES_DATE to the event timestamp + 1 second?